parity-crypto/src/pbkdf2/mod.rs

```diff
@@ -14,21 +14,20 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
+use rhmac;
+use rpbkdf2;
 use std::num::NonZeroU32;
 
-use ring;
-
 pub struct Salt<'a>(pub &'a [u8]);
 pub struct Secret<'a>(pub &'a [u8]);
 
 pub fn sha256(iter: NonZeroU32, salt: Salt, sec: Secret, out: &mut [u8; 32]) {
-	ring::pbkdf2::derive(&ring::digest::SHA256, iter, salt.0, sec.0, &mut out[..])
```

Comment:
If we're getting rid of `ring`, we might as well remove `NonZeroU32`, I guess it's going to be a breaking change anyway.

Reply:
Why get rid of the `NonZeroU32`? It's an "extra protection".

Reply:
Yes, it adds some protection, but we're mostly using constants for iterations. The API for creating `NonZeroU32` is either [unsafe](https://doc.rust-lang.org/std/num/struct.NonZeroU32.html#method.new_unchecked) or [clumsy to use](https://github.com/paritytech/parity-ethereum/pull/10262/files) (waiting for stable const eval).

Reply:
I'm not against removing it. On the other hand, it may not be worth having one single breaking change just for this, while every other change is backwards compatible.

Reply:
Agree, but [we're using ring for pub error type](https://github.com/paritytech/parity-common/blob/e67f50caf03701cee2c3fdc659b1f8b9608363a8/parity-crypto/src/error.rs#L93), so it will be a breaking change.

Reply:
Removed the `NonZeroU32`.